submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/nfs/NfsFtpService.kt

```diff
@@ -53,5 +55,41 @@ class NfsFtpService(
         return nfsFile
     }
 
-    private fun getPublicFolder(relPath: String): Path = FileUtils.getOrCreateFolder(folderResolver.getPublicSubFolder(relPath), RWXR_XR_X)
+    override suspend fun suppressSubmissionFile(
+        sub: ExtSubmissionInfo,
+        file: ExtFile,
+    ): ExtFile =
+        withContext(Dispatchers.IO) {
+            val nfsFile = file as NfsFile
+            val publicSubFolder = getPublicFolder(sub.relPath)
+            val privateSubFolder = folderResolver.getPrivateSubFolder(sub.secretKey, sub.relPath)
+
+            when (releaseMode) {
+                MOVE -> moveSuppress(nfsFile, privateSubFolder)
+                HARD_LINKS -> hardLinkSuppress(nfsFile, publicSubFolder)
```

Comment:
you are only using the publicSubFolder in HARD_LINKS mode and privateSubFolder MOVE mode. I would suggest move variables declaration there rather that populate this scope with them where they are not needed.

Reply:
Nice catch 👍